sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CaseDirectorySteps.java

```diff
@@ -704,6 +657,23 @@ public CaseDirectorySteps(
     And(
         "I apply Day filter {string} on Case directory page",
         (String day) -> webDriverHelpers.selectFromCombobox(CASE_DAY_FILTER, day));
+
+    And(
+        "I check that only cases with fulfilled reference definition are being shown in Cases directory",
+        () -> {
+          Integer totalResults =
+              Integer.parseInt(webDriverHelpers.getTextFromPresentWebElement(TOTAL_CASES_COUNTER));
+          for (int i = 0; totalResults > 0 && i < totalResults && i < 5; i++) {
```

Comment:
Since the reference definition isn't shown in the table, but requires opening the Edit Case page, we're only checking up to 5 results that way.

This is because:

- If there are dozens, or even hundreds of search results, checking them all would take a ridiculous amount of time
- The page does not load the entire table, but rather builds it dynamically, which leads to flakiness when trying to access more than 10-15 cells down the line